sklearn/linear_model/_quantile.py

```diff
@@ -246,14 +246,16 @@ def fit(self, X, y, sample_weight=None):
             c[0] = 0
             c[n_params] = 0
 
-        if sparse.issparse(X):
-            if self.solver not in ["highs-ds", "highs-ipm", "highs"]:
+        if sparse.issparse(X) or self.solver in ["highs", "highs-ds", "highs-ipm"]:
+            if self.solver not in ["highs", "highs-ds", "highs-ipm"]:
```

Comment:
I think it would be a slightly nicer to do this check before computing `c`, so we fail earlier. In other words, include the following at around [L195](https://github.com/scikit-learn/scikit-learn/blob/9daed7f8970c25afaa54f671815afd75605acf26/sklearn/linear_model/_quantile.py#L195) where we validate the solver:

```python
if sparse.issparse(X) and self.solver not in ["highs", "highs-ds", "highs-ipm"]:
    raise ValueError(...)
```

This way we only need to condition only `highs` here.